lib/ansible/modules/network/nxos/nxos_interface.py

```diff
@@ -277,70 +277,74 @@ def get_interface(intf, module):
+                    key_map.update(mode_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    temp_dict = apply_value_map(mode_value_map, temp_dict)
+                    interface.update(temp_dict)
+
+                elif intf_type == 'svi':
+                    key_map.update(svi_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    interface.update(temp_dict)
+                    attributes = get_manual_interface_attributes(intf, module)
+                    interface['admin_state'] = str(attributes.get('admin_state',
+                                                                  'nxapibug'))
+                    interface['description'] = str(attributes.get('description',
+                                                                  'nxapi_bug'))
+                    command = 'show run interface {0}'.format(intf)
+                    body = execute_show_command(command, module)[0]
+                    if 'ip forward' in body:
+                        interface['ip_forward'] = 'enable'
+                    else:
+                        interface['ip_forward'] = 'disable'
+                    if 'fabric forwarding mode anycast-gateway' in body:
+                        interface['fabric_forwarding_anycast_gateway'] = True
+                    else:
+                        interface['fabric_forwarding_anycast_gateway'] = False
+
+                elif intf_type == 'loopback':
+                    key_map.update(base_key_map)
+                    key_map.pop('admin_state')
+                    key_map.update(loop_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    if not temp_dict.get('description'):
+                        temp_dict['description'] = "None"
+                    interface.update(temp_dict)
+
+                elif intf_type == 'management':
+                    key_map.update(base_key_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    interface.update(temp_dict)
+
+                elif intf_type == 'portchannel':
+                    key_map.update(base_key_map)
+                    key_map.update(mode_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    temp_dict = apply_value_map(mode_value_map, temp_dict)
+                    if not temp_dict.get('description'):
+                        temp_dict['description'] = "None"
+                    interface.update(temp_dict)
+
+                elif intf_type == 'nve':
+                    key_map.update(base_key_map)
+                    temp_dict = apply_key_map(key_map, interface_table)
+                    if not temp_dict.get('description'):
+                        temp_dict['description'] = "None"
+                    interface.update(temp_dict)
+        except TypeError:
+            return ""
```

Comment:
Empty string is usually not a good value to return as a sentinel.  it's probably better return an empty dict (since the other case returns a dict)  You could also catch this exception higher up if you don't actually know how to recover here. From my brief outsiders look at the code it seems like that might the case (empty string might be a sentinel value to represent an error rather than good data to be used by the calling functions.  In python [and most languages that have exceptions] exceptions a re the way to represent error conditions, not sentinel values.)

Handling the error higher up the call stack would also get rid of a style problem.  it's usually a sign of poor code if a try: except spans more than a few lines (Yes, there are many places in Ansible where try: except blocks cover many lines... Those should be refactored at some point in the future when we have time).  This is because you might anticipate catching TypeError being emitted from one function call but not realize that typeerror could be emitted by a different function call which  your error handler cannot recover from.